momento-sdk/src/intTest/java/momento/sdk/retry/FixedCountRetryStrategyIntegTest.java

```diff
@@ -0,0 +1,109 @@
+            .returnError(MomentoErrorCode.SERVER_UNAVAILABLE.name())
+            .errorRpcList(Collections.singletonList(MomentoRpcMethod.GET.getRequestName()))
+            .build();
+
+    withCacheAndCacheClient(
+        config -> config,
+        testRetryMetricsMiddlewareArgs,
+        (cacheClient, cacheName) -> {
+          assertThat(cacheClient.get(cacheName, "key"))
+              .succeedsWithin(FIVE_SECONDS)
+              .asInstanceOf(InstanceOfAssertFactories.type(GetResponse.Error.class))
+              .satisfies(
+                  error ->
+                      assertThat(error.getErrorCode())
+                          .isEqualTo(MomentoErrorCode.SERVER_UNAVAILABLE));
+
+          assertThat(testRetryMetricsCollector.getTotalRetryCount(cacheName, MomentoRpcMethod.GET))
+              .isEqualTo(3);
+        });
+  }
+
+  @Test
+  void testNonRetryEligibleApi_shouldMakeNoAttempts_WhenFullNetworkOutage() throws Exception {
+    TestRetryMetricsMiddlewareArgs testRetryMetricsMiddlewareArgs =
+        new TestRetryMetricsMiddlewareArgs.Builder(
+                logger, testRetryMetricsCollector, UUID.randomUUID().toString())
+            .returnError(MomentoErrorCode.SERVER_UNAVAILABLE.name())
+            .errorRpcList(Collections.singletonList(MomentoRpcMethod.INCREMENT.getRequestName()))
+            .build();
+
+    withCacheAndCacheClient(
+        config -> config,
+        testRetryMetricsMiddlewareArgs,
+        (cacheClient, cacheName) -> {
+          assertThat(cacheClient.increment(cacheName, "key", 1))
+              .succeedsWithin(FIVE_SECONDS)
+              .asInstanceOf(InstanceOfAssertFactories.type(IncrementResponse.Error.class))
+              .satisfies(
+                  error ->
+                      assertThat(error.getErrorCode())
+                          .isEqualTo(MomentoErrorCode.SERVER_UNAVAILABLE));
+
+          assertThat(
+                  testRetryMetricsCollector.getTotalRetryCount(
+                      cacheName, MomentoRpcMethod.INCREMENT))
+              .isEqualTo(0);
+        });
+  }
+
+  @Test
+  void testRetryEligibleApi_shouldMakeLessThanMaxAttempts_WhenTemporaryNetworkOutage()
+      throws Exception {
+    TestRetryMetricsMiddlewareArgs testRetryMetricsMiddlewareArgs =
+        new TestRetryMetricsMiddlewareArgs.Builder(
+                logger, testRetryMetricsCollector, UUID.randomUUID().toString())
+            .returnError(MomentoErrorCode.SERVER_UNAVAILABLE.name())
+            .errorRpcList(Collections.singletonList(MomentoRpcMethod.GET.getRequestName()))
+            .errorCount(2)
+            .build();
+
+    withCacheAndCacheClient(
+        config -> config,
+        testRetryMetricsMiddlewareArgs,
+        (cacheClient, cacheName) -> {
+          assertThat(cacheClient.get(cacheName, "key"))
+              .succeedsWithin(FIVE_SECONDS)
+              .extracting(response -> (GetResponse.Miss) response)
+              .isNotNull();
```

Comment:
You can directly assert that the response is an instance of miss instead of implicitly casing it:
```suggestion
              .isInstanceOf(GetResponse.Miss.class);
```